firmware_configurations/klipper/IamPerrine/SKR_Pancake_Config.txt

```diff
@@ -0,0 +1,678 @@
+
+[gcode_macro G32]
+gcode:
+   NOZZLE_CLEAN
+   QUAD_GANTRY_LEVEL
+   NOZZLE_CLEAN
+   G28 Z0
+   G0 X175 Y175 Z10 F6000
+
+[gcode_macro PRINT_START]
+#   Use PRINT_START for the slicer starting script - please customize for your slicer of choice
+gcode:
+    G28 xyz ; go home
+	G32
+	G0 Y0 X0
+	G1 Z0.2 F500.0
+	G92 E0.0
+	
+	BED_MESH_PROFILE LOAD=hot ; Load Saved Mesh
+
+	G1 X80 E10 F500.0 ; intro line
+	G1 Z0.2
+
+	G92 E0.0
+
+	
+[gcode_macro PRINT_END]
+#   Use PRINT_END for the slicer ending script - please customize for your slicer of choice
+gcode:
+   M400                           ; wait for buffer to clear
+   G92 E0                         ; zero the extruder
+   G1 E-4.0 F3600                 ; retract filament
+   G91                            ; relative positioning
+   G0 Z1.00 X2.0 Y2.0 F20000    ; move nozzle to remove stringing
+   TURN_OFF_HEATERS
+   M107                           ; turn off fan
+   G1 Z5 F3000                    ; move nozzle up 5mm
+   G90                            ; absolute positioning
+   G0  X175 Y350 F3600            ; park nozzle at rear
+   M117 Finished!                 ; display message
+
+[gcode_macro UNLOAD_FILAMENT]
+#   Extrusion lengths must be adjusted for your particular configuration before use
+gcode:
+   M83                            ; set extruder to relative
+   G1 E10 F300                    ; extrude a little to soften tip
+   G1 E-600 F1800                 ; retract filament completely
+   M82                            ; set extruder to absolute
+
+[gcode_macro LOAD_FILAMENT]
+#   Extrusion lengths must be adjusted for your particular configuration before use
+gcode:
+   M83                            ; set extruder to relative
+   G1 E300 F1800                  ; quickly load filament to down bowden
+   G1 E200 F300                    ; slower extrusion for hotend path
+   G1 E-5 F150                    ; prime nozzle with filament
+   M82                            ; set extruder to absolute
+
+[gcode_macro NOZZLE_BRUSH]
+gcode:
+    M117 Nozzle brushing...
+    G0 X287 Y350 F6000
+    G0 Z-1
+    G0 X327
+    G0 X287
+    G0 X327
+    G0 X287
+    G0 X327
+    G0 X287
+    G0 Z10
+
+[gcode_macro NOZZLE_CLEAN]
+gcode:
+    M117 Cleaning Nozzle...
+    G0 Z10
+    nozzle_brush
+    G0 Z10
+    M117
+
+#Pause/Resume Functionality
+[pause_resume]
+
+#Filament Sensor Testing.
+[filament_switch_sensor test]
+pause_on_runout: True
+switch_pin: !P1.25
+runout_gcode:
+ M118 Runout Event Detected
+ M600 X50 Y10 Z20
+ 
+[gcode_macro M600]
+default_parameter_X: 50
+default_parameter_Y: 10
+default_parameter_Z: 10
+gcode:
+ G91
+ G1 Z{Z}
+ G90
+ G1 X{X} Y{Y} F3000
+ G0 E-100 F4800
+	
+# Sensor Types
+#   "EPCOS 100K B57560G104F"
+#   "ATC Semitec 104GT-2"
+#   "NTC 100K beta 3950"
+#   "Honeywell 100K 135-104LAG-J01"
+#   "NTC 100K MGB18-104F39050L32" (Keenovo Heater Pad)
+#   "AD595"
+#   "PT100 INA826"
+
+#*# <---------------------- SAVE_CONFIG ---------------------->
```

Comment:
This block will likely not conform to any other printer but your own. If someone chooses to use this config, they might miss that this block is present and get unexpected behavior. I suggest to remove them.